.gitmodules

```diff
@@ -11,3 +11,6 @@
 [submodule "gtda/externals/eigen"]
 	path = gtda/externals/eigen
 	url = https://gitlab.com/libeigen/eigen
+[submodule "gtda/externals/pybind11"]
+	path = gtda/externals/pybind11
+	url = https://github.com/pybind/pybind11
```

Comment:
Shouldn't we fix the branch too?

Reply:
Git submodules works like this:

1. You add a submodule at some point where the submoudle is at a specific commit
2. You keep using this module at a specific commit, at the same time the developers of the submodule continue to work on it.
3. The changes they do on the submodule are not forwarded to the version you use in order to prevent breaking a functional system
4. At some point you want to integrate the latest changes from the submodule, to do that, you need to pull changes, and push the commit version to the repository.

If you check [here](https://github.com/giotto-ai/giotto-tda/tree/master/gtda/externals) you can see that the submodules have a hash following their name, this correspond to the commit they will checkout (you can look at our hera version and the master in the hera repository are not the same). And this commit does not always correspond to the latest changes

With this method from time to time you can pull latest changes and you'll need to push to the server the version you want to checkout when downloading submodules.

One thing that you made me think, for developers, when integrating this change to their version, they shall delete the `gtda/externals/pybind11` folder in order to add the submodule correctly. This only affects previously developpers.

Reply:
Thanks for the explanation @monkeybreaker, but I still don't understand why we're not fixing the v2.5 branch, as we said we would. I think it can be done in this file, no?

Reply:
We do this right above, with GUDHI.

Reply:
Ah !
Well we could of course, I just took the latest changes last Friday, would you prefer latest stable release of `pybind11` (v2.5 from 31 March) ?

Reply:
I suppose since we are fixing a commit, issues will only occur those rare times when we consciously decide to pull from `pybind11` master. So it's OK not to fix a branch!